core/app/models/spree/product/slugs.rb

```diff
@@ -57,8 +57,6 @@ def ensure_slug_is_unique(candidate_slug)
       def slug_candidates
         if defined?(:deleted_at) && deleted_at.present?
           [
-            ['deleted', :name],
-            ['deleted', :name, :sku],
             ['deleted', :name, :uuid_for_friendly_id]
```

Comment:
We always need to add the uuid since deleted products are not in the scope visible when generating slugs for deleted products.